src/QMCWaveFunctions/SpinorSet.cpp

```diff
@@ -178,6 +178,113 @@ void SpinorSet::evaluate_notranspose(const ParticleSet& P,
   }
 }
 
+void SpinorSet::mw_evaluate_notranspose(const RefVectorWithLeader<SPOSet>& spo_list,
+                                        const RefVectorWithLeader<ParticleSet>& P_list,
+                                        int first,
+                                        int last,
+                                        const RefVector<ValueMatrix_t>& logdet_list,
+                                        const RefVector<GradMatrix_t>& dlogdet_list,
+                                        const RefVector<ValueMatrix_t>& d2logdet_list) const
+{
+  auto& spo_leader = spo_list.getCastedLeader<SpinorSet>();
+  auto& P_leader   = P_list.getLeader();
+  assert(this == &spo_leader);
+
+  IndexType nw    = spo_list.size();
+  IndexType nelec = P_leader.getTotalNum();
+
+  SPOSet& up_spo_leader = *(spo_leader.spo_up);
+  SPOSet& dn_spo_leader = *(spo_leader.spo_dn);
+  RefVectorWithLeader<SPOSet> up_spo_list(up_spo_leader);
+  RefVectorWithLeader<SPOSet> dn_spo_list(dn_spo_leader);
+  up_spo_list.reserve(nw);
+  dn_spo_list.reserve(nw);
+  up_spo_list.clear();
+  dn_spo_list.clear();
+
+  std::vector<ValueMatrix_t> mw_up_logdet, mw_dn_logdet;
+  std::vector<GradMatrix_t> mw_up_dlogdet, mw_dn_dlogdet;
+  std::vector<ValueMatrix_t> mw_up_d2logdet, mw_dn_d2logdet;
+  mw_up_logdet.reserve(nw);
+  mw_dn_logdet.reserve(nw);
+  mw_up_dlogdet.reserve(nw);
+  mw_dn_dlogdet.reserve(nw);
+  mw_up_d2logdet.reserve(nw);
+  mw_dn_d2logdet.reserve(nw);
+  mw_up_logdet.clear();
+  mw_dn_logdet.clear();
+  mw_up_dlogdet.clear();
+  mw_dn_dlogdet.clear();
+  mw_up_d2logdet.clear();
+  mw_dn_d2logdet.clear();
```

Comment:
No need to clear. `mw_up_logdet` defined in the scope start with size 0.

Reply:
fixed